client/my-sites/upgrades/domain-management/list/index.jsx

```diff
@@ -46,26 +65,183 @@ const List = React.createClass( {
 					path={ this.props.context.path }
 					cart={ this.props.cart }
 					selectedSite={ this.props.selectedSite } />
-
 				{ this.domainWarnings() }
 
 				<SectionHeader label={ headerText }>
-					{ this.addDomainButton() }
+					{ this.headerButtons() }
 				</SectionHeader>
 
 				<div className="domain-management-list__items">
+					{ this.notice() }
 					{ this.listItems() }
 				</div>
 			</Main>
 		);
 	},
 
+	hideNotice() {
+		this.setState( { notice: null } );
+	},
+
+	notice() {
+		const { notice } = this.state;
+		if ( ! notice ) {
+			return null;
+		}
+
+		const { type, error, domainName } = notice,
+			props = {
+				className: 'domain-management-list__notice',
+				showDismiss: false,
+				onDismissClick: this.hideNotice
+			};
+
+		let errorMessage;
+
+		switch ( type ) {
+			case PRIMARY_DOMAIN_CHANGE_SUCCESS:
+				Object.assign( props, {
+					text: <span key="text">{ this.translate( 'Primary Domain changed: All domains will redirect to' +
+						' {{em}}%(domainName)s' +
+						'{{/em}}.', { args: { domainName }, components: { em: <em/> } } ) }</span>,
```

Comment:
This seems like a weird place to break the string? `{{em}}%(domainName)s{{/em}}.` can definitely be on one line.